File: packages/cli/src/commands/stop.ts
import { getConfig } from '../utils/config.js';
import { getContainerStatus, removeContainer, stopContainer } from '../utils/docker.js';
import { log } from '../utils/ui.js';

export type StopOptions = {
  remove?: boolean;
};

export async function stopCommand(options: StopOptions): Promise<void> {
  const { containerName } = getConfig();
  const status = await getContainerStatus(containerName);

  if (!status.exists) {
    log.warn('Container does not exist.');
    return;
  }

  if (status.running) {
    await stopContainer(containerName);
    log.success('Container stopped.');
  } else {
    log.warn('Container is already stopped.');
  }

  if (options.remove) {
    await removeContainer(containerName);
    log.success('Container removed.');
  }
}

File: packages/cli_native/src/commands/stop.rs
use anyhow::Result;

use crate::utils::{
    config::get_config,
    docker::{get_container_status, remove_container, stop_container},
    ui::LOG,
};

pub async fn stop_command(remove: bool) -> Result<()> {
    let config = get_config();
    let container_name = &config.container_name;
    let status = get_container_status(container_name).await?;

    if !status.exists {
        LOG.warn("Container does not exist.");
        return Ok(());
    }

    if status.running {
        stop_container(container_name).await?;
        LOG.success("Container stopped.");
    } else {
        LOG.warn("Container is already stopped.");
    }

    if remove {
        remove_container(container_name).await?;
        LOG.success("Container removed.");
    }

    Ok(())
}
